lib/command.c

```diff
@@ -2668,6 +2671,12 @@ int cmd_banner_motd_file(const char *file)
 	return success;
 }
 
+int cmd_banner_motd_line(const char *line)
+{
+	host.motd = line;
```

Comment:
Don't you need to free the previous motd before assigning a new one?

Reply:
Looking what banner_motd_default_cmd does, seems not necessary:
```
DEFUN (banner_motd_default,
       banner_motd_default_cmd,
       "banner motd default",
       "Set banner string\n"
       "Strings for motd\n"
       "Default string\n")
{
	host.motd = default_motd;
	return CMD_SUCCESS;
}
```

Reply:
That's because `host.motd` is either NULL or points to `default_motd` in the original code.

Your new command creates a new string in the heap using `argv_concat()` and assigns that to `host.motd` (which is a `const char *` pointer). Now `host.motd` can contain either a static or a dynamically allocated string, which isn't good.

I'd suggest you to do one of the following:
* Change `host.motd` to `char *` and use `XSTRDUP` whenever assigning `default_motd` to it. Also, always release the previous value of `host.motd` (if any) before assigning a new one;
* Create a new `motdline` field in the `host` structure similar to `motdfile` and adapt the code accordingly.

Reply:
fixed